src/streamlink/plugins/pandalive.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import re
+
+from streamlink.plugin import Plugin, PluginError, pluginmatcher
+from streamlink.plugin.api import validate
+from streamlink.stream.hls import HLSStream
+
+log = logging.getLogger(__name__)
+
+
+@pluginmatcher(re.compile(
+    r"https?://(?:www\.)?pandalive\.co\.kr/"
+))
+class Pandalive(Plugin):
+    API_URL = "https://api.pandalive.co.kr/v1/live/play"
+
+    _room_id_re = re.compile(r"roomid\s*=\s*String\.fromCharCode\((.*)\)")
+
+    def _get_media_code(self):
+        try:
+            return self.session.http.get(self.url, schema=validate.Schema(
+                validate.parse_html(),
+                validate.xml_xpath_string(".//script[contains(text(), 'roomid')]/text()"),
+                str,
+                validate.transform(self._room_id_re.search),
+                validate.get(1),
+                validate.transform(lambda s: "".join(map(lambda c: chr(int(c)), s.split(",")))),
+            ))
+        except PluginError:
+            pass
+
+        return None
+
+    def _get_streams(self):
+        media_code = self._get_media_code()
+
+        if not media_code:
+            return
+
+        log.debug(f"Media code: {media_code}")
+
+        data = {
+            "action": "watch",
+            "mediaCode": media_code,
+        }
+        json = self.session.http.post(self.API_URL, data=data, schema=validate.Schema(
+            validate.parse_json(), {
+                validate.optional("media"): {
+                    "title": str,
+                    "userId": str,
+                    "userNick": str,
+                    "isLive": bool,
+                },
+                validate.optional("PlayList"): {
+                    "hls2": [{
+                        "url": validate.url(),
+                    }],
+                },
+                "result": bool,
+            },
+        ))
+
+        if not json["result"] or not json["media"]["isLive"]:
+            log.error("The broadcast has ended")
+            return
+
+        self.author = f"{json['media']['userNick']} ({json['media']['userId']})"
+        self.title = f"{json['media']['title']}"
+
+        for stream in json["PlayList"]["hls2"]:
+            yield from HLSStream.parse_variant_playlist(self.session, stream["url"]).items()
```

Comment:
> I've only used hls2 for the URLs for the time being, as all point to the same 720p-only variant playlist

Please just return one (variant) playlist then.

Reply:
I did wonder about that.  The only reason I was getting both was to allow for the site at some point in the future actually returning a 360p variant playlist for the 360p URL.